File: algo/src/main/java/com/hartwig/actin/algo/evaluation/general/IsInvolvedInStudyProcedures.java
package com.hartwig.actin.algo.evaluation.general;

import com.hartwig.actin.PatientRecord;
import com.hartwig.actin.algo.datamodel.Evaluation;
import com.hartwig.actin.algo.datamodel.EvaluationResult;
import com.hartwig.actin.algo.datamodel.ImmutableEvaluation;
import com.hartwig.actin.algo.evaluation.EvaluationFunction;

import org.jetbrains.annotations.NotNull;

public class IsInvolvedInStudyProcedures implements EvaluationFunction {

    IsInvolvedInStudyProcedures() {
    }

    @NotNull
    @Override
    public Evaluation evaluate(@NotNull PatientRecord record) {
        return ImmutableEvaluation.builder().result(EvaluationResult.NOT_EVALUATED).build();
    }
}

File: algo/src/main/kotlin/com/hartwig/actin/algo/evaluation/general/IsInvolvedInStudyProcedures.kt
package com.hartwig.actin.algo.evaluation.general

import com.hartwig.actin.algo.evaluation.EvaluationFactory
import com.hartwig.actin.algo.evaluation.EvaluationFunction
import com.hartwig.actin.datamodel.PatientRecord
import com.hartwig.actin.datamodel.algo.Evaluation

class IsInvolvedInStudyProcedures : EvaluationFunction {
    
    override fun evaluate(record: PatientRecord): Evaluation {
        return EvaluationFactory.fail("Assumed that there is no involvement with the trial of interest")
    }
}